### PR TITLE
Noise level estimation for a single image. 

### PR DESCRIPTION
**Pull request by MSKirk**
A direct translation of NoiseLevel.m to estimate AWGN using weakly textured patches.

### CODE REVIEW

**Comment by DanRyanIrish** [comment]
```NoiseLevel_test.py``` should be ```test_noiselevel.py``` or ```test_noise_level.py```

File: sunkit_image/utils/NoiseLevel.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
+import numpy as np
+from scipy.stats import gamma
+from scipy.ndimage import correlate
+from skimage.util import view_as_windows
+
```
Comment (Cadair) [inline sunkit_image/utils/NoiseLevel.py]: we should have an `__all__` list with the function / class names here.

File: sunkit_image/utils/NoiseLevel.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+class NoiseLevelEstimation:
+    def __init__(self, img, patchsize=7, decim=0, conf=1-1E-6, itr=3):
+        """
+        NoiseLevel estimates noise level of input single noisy image.
+
+        Input parameters:
```
Comment (Cadair) [inline sunkit_image/utils/NoiseLevel.py]: This should use numpydoc styling, i.e.
```
Parameters
----------
```

same as the other section headings :)

File: sunkit_image/utils/NoiseLevel.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+        Example:
+            estimate = NoiseLevelEstimation(noisy_image_array, patchsize=11, itr=10)
+
+        Python Version: 20180718
+        Python Author: M. Kirk
+
+        Translated from Noise Level Estimation Matlab code: noiselevel.m
+
+        noiselevel.m Copyright (C) 2012-2015 Masayuki Tanaka
```
Comment (Cadair) [inline sunkit_image/utils/NoiseLevel.py]: What licence is the original version under?

File: sunkit_image/utils/NoiseLevel.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+            itr (optional): number of iteration. (default: 3)
+
+        Calculated parameters:
+            nlevel: estimated noise levels.
+            th: threshold to extract weak texture patches at the last iteration.
+            num: number of extracted weak texture patches at the last iteration.
+            mask: weak-texture mask. 0 and 1 represent non-weak-texture and weak-texture regions, respectively
+
+        Example:
+            estimate = NoiseLevelEstimation(noisy_image_array, patchsize=11, itr=10)
```
Comment (Cadair) [inline sunkit_image/utils/NoiseLevel.py]: Add an example of accessing the calculate parameters?

File: sunkit_image/utils/NoiseLevel.py (hunk 3 above)
Comment (MSKirk) [reply sunkit_image/utils/NoiseLevel.py]: From how I read it, I think we can include it as long as we add in the original license file.

Comment (Cadair) [reply sunkit_image/utils/NoiseLevel.py]: looks like BSD 2-clause to me, I agree that we can use it.

**Comment by wafels** [comment]
I think this should live in its own subdirectory, something like /noise.  Is there a descriptive name for the method we could use instead of "NoiseLevel".  Also, what does AWGN mean?

**Comment by MSKirk** [comment]
"Additive white Gaussian noise (AWGN) is a basic noise model used in Information theory to mimic the effect of many random processes that occur in nature."

**Comment by nabobalis** [comment]
Would it be ok If I rebased this and addressed the comments?

**Comment by MSKirk** [comment]
@nabobalis Go for it! I haven't had the time to look at it recently, but I intend to get back to that project soon.

**Comment by nabobalis** [comment]
My documentation will need a read over to make sure it makes any sense.

**Comment by nabobalis** [comment]
Thanks @MSKirk!

**Comment by MSKirk** [comment]
Thanks @nabobalis for pushing this across the finish line.

**Comment by nabobalis** [comment]
You did the hard work, I just tidied some strings up!

**Pull request merged**